test/skimNtuple2016_HHbtag.cpp

```diff
@@ -1465,15 +1381,15 @@ int main (int argc, char** argv)
       int dauType = theBigTree.particleType->at(idau);
       if (oph.isMuon(dauType))
       {
-	bool passMu   = oph.muBaseline (&theBigTree, idau, 19., 2.3, 0.15, OfflineProducerHelper::MuTight, string("All") , (DEBUG ? true : false));
+	bool passMu   = oph.muBaseline (&theBigTree, idau, 20., 2.4, 0.15, OfflineProducerHelper::MuTight, string("All") , (DEBUG ? true : false));
```

Comment:
We need to verify that trigger and id+iso scale factors exist up to eta 2.4

Reply:
[Twiki](https://twiki.cern.ch/twiki/bin/view/CMS/MuonUL2018#Trigger_efficiency_AN1)

- [X] UL Trigger SFs: exist up to 2.4 for all years ([presentation](https://indico.cern.ch/event/1080036/contributions/4542924/attachments/2318322/3947065/210928_ULTriggerSF_kHwang.pdf), [json with SFs](https://gitlab.cern.ch/cms-muonPOG/muonefficiencies/-/blob/master/Run2/UL/2018/2018_trigger/Efficiencies_muon_generalTracks_Z_Run2018_UL_SingleMuonTriggers_schemaV2.json#L38)). They include SFs for both ```HLT_IsoMu24``` and ```HLT_IsoMu27```.

I did test what happens with our current Legacy weights when we ask for a particular SF label for 2018 data (```Muon_Run2018_IsoMu20.root``` file):

```
root [10] etaBinsH->GetXaxis()->FindFixBin(2.3)
(int) 4
root [11] etaBinsH->GetXaxis()->FindFixBin(2.4)
(int) 4
root [12] etaBinsH->GetXaxis()->FindFixBin(2.5)
(int) 5
root [13] etaBinsH->GetXaxis()->FindFixBin(999999)
(int) 5
root [14] etaBinsH->GetXaxis()->GetBinLabel(3)
(const char *) "Eta1p2to2p1"
root [15] etaBinsH->GetXaxis()->GetBinLabel(4)
(const char *) "EtaGt2p1"
root [16] etaBinsH->GetXaxis()->GetBinLabel(5)
(const char *) ""
```

We currently cut on 2.4, so between 2.1 and 2.4 our muons will be assigned the weight stored in "EtaGt2p1". This is why we get no error.
For 2016 (file ```Muon_Mu19leg_2016BtoH_eff.root```), you only have three labels: "EtaLt0p9", "Eta0p9to1p2" and "Eta1p2to2p1", hence the error. 

To recap, the issue should be gone once we update to UL SFs.

- [X] UL Iso/Id SFs: the versions we currently have are already Ultra-Legacy. Both ID and ISO SFs for muons go up to 2.4, so all good!

Reply:
Thanks for checking Bruno. This means that before running the 2016 production we will have to either switch to the UL SF or temporarily revert the cut to 2.1

Reply:
Yes. I suggest not touching the code (to save your time!), and I will do a PR with the new SFs asap for the three years.

Reply:
Ok, sounds good to me! Thanks a lot!